exposed-money/src/main/kotlin/org/jetbrains/exposed/sql/money/CurrencyColumnType.kt

```diff
@@ -12,23 +14,62 @@ import javax.money.Monetary
  * @author Vladislav Kisel
  */
 @Suppress("MagicNumber")
-class CurrencyColumnType : VarCharColumnType(3) {
+class CurrencyColumnType : ColumnType<CurrencyUnit>() {
 
-    override fun notNullValueToDB(value: Any): Any {
-        return when (value) {
-            is String -> value
-            is CurrencyUnit -> value.currencyCode
-            else -> error("Unexpected value: $value of ${value::class.qualifiedName}")
+    override fun sqlType(): String = currentDialect.dataTypeProvider.varcharType(colLength)
+
+    override fun validateValueBeforeUpdate(value: CurrencyUnit?) {
+        if (value is CurrencyUnit) {
+            val valueLength = value.currencyCode.codePointCount(0, value.currencyCode.length)
+            require(valueLength <= colLength) {
+                "Value can't be stored to database column because exceeds length ($valueLength > $colLength)"
+            }
         }
     }
 
-    override fun valueFromDB(value: Any): Any {
+    override fun valueFromDB(value: Any): CurrencyUnit {
         return when (value) {
             is CurrencyUnit -> value
             is String -> Monetary.getCurrency(value)
             else -> valueFromDB(value.toString())
         }
     }
+
+    override fun notNullValueToDB(value: CurrencyUnit): Any = value.currencyCode
+
+    override fun nonNullValueToString(value: CurrencyUnit): String = buildString {
+        append('\'')
+        append(escape(value.currencyCode))
+        append('\'')
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+        if (!super.equals(other)) return false
+
+        other as VarCharColumnType
+
+        return colLength == other.colLength
+    }
+
+    override fun hashCode(): Int {
+        var result = super.hashCode()
+        result = 31 * result + colLength
+        return result
+    }
+
+    private fun escape(value: String): String = value.map { charactersToEscape[it] ?: it }.joinToString("")
+
+    companion object {
+        private const val colLength = 3
+
+        private val charactersToEscape = mapOf(
+            '\'' to "\'\'",
+            '\r' to "\\r",
+            '\n' to "\\n"
+        )
```

Comment:
Would it be possible to extract this property (and also the logic in `nonNullValueToString()`) since it's duplicated by a few of the string column types? Or at least just for the common types that use it in `ColumnType.kt`.